tiledb/sm/filesystem/s3.cc

```diff
@@ -28,6 +28,15 @@
  * @section DESCRIPTION
  *
  * This file implements the S3 class.
+ *
+ * ============================================================================
+ * Notes on implementation limitations:
+ *
+ * When using MinIO object storage, files may mask Array directories
+ * if they are of the same name. As such, the ArrayDirectory may filter out
+ * non-empty directories. Users should be aware of this limitation and attempt
+ * to manually maintain their working directory to avoid test failures.
+ * ============================================================================
```

Comment:
I don't think we need this note now that we're throwing an exception that calls out the issue specifically.

Reply:
cc @ihnorton 
I still think the comment is helpful but am not tied to it

Reply:
That's enough for me. Leave it be then.